internal/internal_task_pollers.go

```diff
@@ -632,6 +644,23 @@ func (wtp *workflowTaskPoller) getNextPollRequest() (request *workflowservice.Po
 	}
 }
 
+// Poll the workflow task queue and update the num_poller metric
+func (wtp *workflowTaskPoller) pollWorkflowTaskQueue(ctx context.Context, request *workflowservice.PollWorkflowTaskQueueRequest) (*workflowservice.PollWorkflowTaskQueueResponse, error) {
+	if request.TaskQueue.GetKind() == enumspb.TASK_QUEUE_KIND_NORMAL {
+		wtp.numNormalPollersGauge.Update(float64(atomic.AddInt32(&wtp.numNormalPollers, 1)))
```

Comment:
This is racy. What if `numNormalPollers` is atomically updated twice in one order but gauge `Update` is called twice in a different order. You could have:

```
numNormalPollers=20
numNormalPollers=21
gauge.Update(21)
gauge.Update(20)
```

If you want truly accurate numbers, you must make the gauge update atomic, not just the var. If there is a perf concern, we can discuss accuracy vs contention though if we're willing to accept inaccurate. But I think contention is low enough here, locking the gauge update is reasonable.